packages/iov-tendermint-rpc/src/encodings.ts

```diff
@@ -170,3 +178,55 @@ export class Hex {
     return Encoding.fromHex(hexString);
   }
 }
+
+// Encodings needed for hashing block headers
+// Several of these functions are inspired by https://github.com/nomic-io/js-tendermint/blob/tendermint-0.30/src/
+
+// See https://github.com/tendermint/go-amino/blob/v0.15.0/encoder.go#L193-L195
+export function encodeString(s: string): Uint8Array {
+  const utf8 = toUtf8(s);
+  return Uint8Array.from([utf8.length, ...utf8]);
+}
+
+// See https://github.com/tendermint/go-amino/blob/v0.15.0/encoder.go#L79-L87
+export function encodeInt(n: number): Uint8Array {
+  // tslint:disable-next-line:no-bitwise
+  return n >= 0x80 ? Uint8Array.from([(n & 0xff) | 0x80, ...encodeInt(n >> 7)]) : Uint8Array.from([n & 0xff]);
+}
+
+// See https://github.com/tendermint/go-amino/blob/v0.15.0/encoder.go#L134-L178
+export function encodeTime(time: ReadonlyDateWithNanoseconds): Uint8Array {
+  const milliseconds = time.getTime();
+  const seconds = Math.floor(milliseconds / 1000);
+  const secondsArray = seconds ? [0x08, ...encodeInt(seconds)] : new Uint8Array();
+  const nanoseconds = (time.nanoseconds || 0) + (milliseconds % 1000) * 1e6;
+  const nanosecondsArray = nanoseconds ? [0x10, ...encodeInt(nanoseconds)] : new Uint8Array();
+  return Uint8Array.from([...secondsArray, ...nanosecondsArray]);
+}
+
+// See https://github.com/tendermint/go-amino/blob/v0.15.0/encoder.go#L180-L187
+export function encodeHash(hash: Uint8Array): Uint8Array {
+  // Since we're only dealing with short hashes we don't need a full VarBuffer implementation yet
+  return hash.length ? Uint8Array.from([hash.length, ...hash]) : hash;
```

Comment:
Since `Uint8Array`s are mutable, I'd avoid returning the input object directly. I did not find a way to mess around with this in this case since it only works for the empty array, but in general it would be unexpected if the input data changes when manipulating the encoded data.